### ISSUE TITLE
Modify VIP Pool size (Add/Remove)

### ISSUE THREAD

**Issue by PragyaDw** [Azure/aks-hci#218]
Customers want to have the ability to change the network settings specified at target cluster creation to account for changing numbers of applications and services in a target cluster

In this feature we will allow the changing of the VIP pool as that is the most common one to be under sized. The way this is going to be accomplished to allow customers to add an additional IP address range to the current VIP Pool(s) in the target cluster. From a UX perspective this will be accomplished through a new set of PowerShell commands.

**Comment by eponerine** [comment]
I cannot stress how important and awesome this would be! Folks commonly undersize their VIP pool, especially in POC scenarios. They also don't understand _how_ VIPs are used and can paint themselves in a corner when performing cluster upgrades.

+1 from me!

**Comment by Elektronenvolt** [comment]
Yes, seeing this topic coming up as well. In some situations, it's more convenient to use load balancer instead of clusterIP and creating an ingress. Available IPs are gone fast on-premise.
@eponerine - in which issues you run at cluster upgrades with the VIP pool?

**Comment by eponerine** [comment]
@Elektronenvolt - assume you had small pools for both service IPs and node IPs. When you upgrade AKS-HCI, it spawns new Worker Node VMs (which burns a node IP) and reschedules pods (which may or may not burn service IPs). 

If you were near capacity for either of those pools, scheduling and VM network attach may fail, resulting in the upgrade process failing. The only way to get out of this chicken/egg scenario is to manually delete deployed services (freeing up a VIP) or horizontally scaling down the node pool (freeing up a node IP).

Both are somewhat destructive and unnecessary if we could just modify the IP pool settings on-the-fly.

**Comment by eponerine** [comment]
@Elektronenvolt - I dont have any error messages available without spinning and repro'ing in a lab. But yes, that error message you pasted looks familiar.

**Comment by DarrylvanderPeijl** [comment]
`C:\Program Files\AksHci\kvactl.exe cluster nodepool create --clustername loadcluster --nodepoolconfig                   "c:\ClusterStorage\Volume02\ImageStore\1.0.16.10113\yaml\loadcluster-loadpool1.yaml" --kubeconfig                       "c:\ClusterStorage\Volume02\ImageStore\1.0.16.10113\kubeconfig-mgmt" System.Collections.Hashtable.generic_non_zero 1    [Error: Failed to monitor nodepool deployment: loadpool1-md-6f6fc464fc-jzqb2 hit critical error: failed to create nic   loadpool1-7475b587f7-6t6j8-nic for machine loadpool1-7475b587f7-6t6j8: unable to create VM network interface: failed    to create network interface loadpool1-7475b587f7-6t6j8-nic in resource group clustergroup-loadcluster: rpc error: code  = Unknown desc = Could not find an available address in ip pool of type VM: OutOfCapacity]  `

This looks familiar indeed, glad it's my lab and I can just uninstall AKS Hybrid and start over.......

**Comment by PragyaDw** [comment]
Hi @eponerine and @DarrylvanderPeijl acknowledging! Will bring it back into the feature discussions. Adding @mkostersitz and @scooley for visibility.
